sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -453,8 +453,11 @@ abstract class CastBase extends UnaryExpression with TimeZoneAwareExpression wit
     if (d.isNaN || d.isInfinite) null else (d * MICROS_PER_SECOND).toLong
   }
 
-  // converting seconds to us
-  private[this] def longToTimestamp(t: Long): Long = SECONDS.toMicros(t)
+  // SPARK-31710 converting seconds to us,Add compatibility flag
+  private[this] def longToTimestamp(t: Long): Long = {
+    if ( SQLConf.get.getConf( SQLConf.LONG_TIMESTAMP_CONVERSION_IN_SECONDS )) t * 1000000L
+    else t * 1000L
```

Comment:
There are constants in `DateTimeConstants`. Please, use them.

Reply:
Nice suggestion, i will recheck my code